### PR TITLE
[buildkite] Change correctness tests to use same phones as benchmark

### PR DESCRIPTION
**Pull request by antiagainst**
Also removed disabled tests and tweaked the labels a bit for consistency.

### CODE REVIEW

**Comment by antiagainst** [comment]
Manually kicked off a new test: https://buildkite.com/iree/iree-android-arm64-v8a/builds/3644

**Comment by GMNGeoffrey** [comment]
Looks like the phones don't have ctest

**Comment by GMNGeoffrey** [comment]
Maybe we should expand the number of phones in the lab before shifting everything over to this? Or at least have specific plans to imminently do so

**Review by GMNGeoffrey — CHANGES_REQUESTED** [review]
Oh and thank you for adding the "common" names for these :-D

**Comment by antiagainst** [comment]
> Maybe we should expand the number of phones in the lab before shifting everything over to this? Or at least have specific plans to imminently do so

It will require buying more devices though, which I don't know how long it takes. (For Samsung S20 at least. Pixel 4 we have a bunch.) So I won't gate the switch on that.

**Comment by antiagainst** [comment]
> Looks like the phones don't have ctest

Yup. SSHed in and installed cmake. Now I'm seeing [test failures](https://buildkite.com/iree/iree-android-arm64-v8a/builds/3644#53ebea6c-bfcc-4f7b-b345-0c5909bfbadf). Late now; will take a look tmr.

**Comment by GMNGeoffrey** [comment]
> > Maybe we should expand the number of phones in the lab before shifting everything over to this? Or at least have specific plans to imminently do so
> 
> It will require buying more devices though, which I don't know how long it takes. (For Samsung S20 at least. Pixel 4 we have a bunch.) So I won't gate the switch on that.

Yeah we can shift over the existing Pixel 4s to the new lab. I don't like the name of the queue that the correctness test looks for being "benchmark-android" if this isn't just for benchmarking. How about we just add the benchmark devices to a second queue for testing android? We can have agents in multiple queues. Then we don't need to have all the stuff setup for benchmarking just to do correctness testing (e.g. cooling plates).

**Comment by antiagainst** [comment]
> > > Maybe we should expand the number of phones in the lab before shifting everything over to this? Or at least have specific plans to imminently do so
> > 
> > 
> > It will require buying more devices though, which I don't know how long it takes. (For Samsung S20 at least. Pixel 4 we have a bunch.) So I won't gate the switch on that.
> 
> Yeah we can shift over the existing Pixel 4s to the new lab. I don't like the name of the queue that the correctness test looks for being "benchmark-android" if this isn't just for benchmarking. How about we just add the benchmark devices to a second queue for testing android? We can have agents in multiple queues. Then we don't need to have all the stuff setup for benchmarking just to do correctness testing (e.g. cooling plates).

Sure. Added a separate queue `test-android` and dual registered existing phones. Once we have more devices in the lab then we can deregister the current phones for benchmarking from `test-android` queue.

**Review by hanhanW — APPROVED** [review]

File: build_tools/buildkite/cmake/android/arm64-v8a/pipeline.yml [hunk 1]
```diff
@@ -25,97 +25,44 @@ steps:
...
-    # TODO(#4861): Re-enable when phone is fixed
-    skip: "Phone is borked. See https://github.com/google/iree/issues/4861"
-    commands:
-      - "buildkite-agent artifact download --step build build-artifacts.tgz ./"
-      - "tar xzf build-artifacts.tgz"
-      - "find build-android/ -name '*.cmake' -exec sed -i \"s!\\$IREE_DOCKER_WORKDIR/!\\$PWD/!g\" {} \\;"
-      - "cd build-android/"
-      - "ctest --output-on-failure"
-    agents:
       - "test-android=true"
```
Comment (GMNGeoffrey) [inline build_tools/buildkite/cmake/android/arm64-v8a/pipeline.yml:51]: queue test-android?
```suggestion
      - "queue=test-android"
```

**Review by GMNGeoffrey — APPROVED** [review]

File: build_tools/buildkite/cmake/android/arm64-v8a/pipeline.yml (hunk 1 above)
Comment (antiagainst) [reply build_tools/buildkite/cmake/android/arm64-v8a/pipeline.yml:51]: This phone is managed by @hanhanW. So he needs to update the tag there first. Right now it's still using `test-android=true`. @hanhanW could you add the queue tag?

**Pull request merged**